core/src/smartcontracts/isi/permissions/checks.rs

```diff
@@ -297,3 +297,58 @@ pub fn check_query_in_instruction(
         Instruction::Fail(_) | Instruction::ExecuteTrigger(_) => Ok(()),
     }
 }
+
+/// Check if permission `token` has right parameters according to `definition`.
+///
+/// Requires only one bypass over `token` and `definition` parameters.
+///
+/// # Errors
+/// - If `token` doesn't have any parameter from `definition`
+/// - If `token` has any parameter that is not in `definition`
+pub fn check_permission_token_parameters(
+    token: &PermissionToken,
+    definition: &PermissionTokenDefinition,
+) -> std::result::Result<(), ValidationError> {
+    use iroha_data_model::ValueKind;
+    use itertools::{
+        EitherOrBoth::{Both, Left, Right},
+        Itertools,
+    };
```

Comment:
Sam's signature imports inside functions. Don't mind this style, I just think that we need to amend the style guide.

Reply:
I just think it's better to show where exactly is `itertools` crate used.